astropy/table/np_utils.py

```diff
@@ -134,8 +143,12 @@ def common_dtype(cols):
     uniq_types = set(tuple(issubclass(col.dtype.type, np_type) for np_type in np_types)
                      for col in cols)
     if len(uniq_types) > 1:
-        raise TableMergeError('Columns have incompatible types {0}'
-                              .format([col.dtype.name for col in cols]))
+        # Embed into the exception the actual list of incompatible types.
+        incompat_types = [col.dtype.name for col in cols]
```

Comment:
@bluescarni @astrofrog – I'm getting this error:
```python
>>> from astropy.table import vstack, Table
>>> t1 = Table({'a': [42]})
>>> t2 = Table({'a': ['test']})
>>> vstack([t1, t2])
Traceback (most recent call last):
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev12059-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 470, in get_descrs
    dtype = common_dtype(in_cols)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev12059-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 507, in common_dtype
    raise tme
astropy.table.np_utils.TableMergeError: Columns have incompatible types ['a', 'a']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev12059-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 254, in vstack
    out = _vstack(tables, join_type, col_name_map)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev12059-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 732, in _vstack
    out_descrs = get_descrs(arrays, col_name_map)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev12059-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 475, in get_descrs
    .format(names[0], tme._incompat_types))
astropy.table.np_utils.TableMergeError: The 'a' columns have incompatible types: ['a', 'a']
```

I was expecting the dtypes printed in the exception, not repeatedly the column name.
(column names have to be identical in all tables, so there's no point in reporting them for each table, no?)